test/unit_CartesianProduct.jl

```diff
@@ -227,6 +242,13 @@ for N in [Float64, Float32, Rational{Int}]
     res, w = ⊆(cpa2, cpa1, true)
     @test !⊆(cpa2, cpa1) && !res && w ∈ cpa2 && w ∉ cpa1
 
+    # convert a hyperrectangle to the cartesian product array of intervals
+    # convert a hyperrectangle to a cartesian product of intervals
+    H = Hyperrectangle(N[0, 0, 0], N[1, 1, 1])
+    Hcp = convert(CartesianProductArray{N, Interval{N}}, H)
+    @test Hcp isa CartesianProductArray &&
+          all([Hcpi == Interval(N(-1), N(1)) for Hcpi in array(Hcp)])
```

Comment:
Same as above.
`H = Hyperrectangle(N[0, 1, 2], N[1, 2, 3])` and then `all([array(Hcp)[i] == Interval(N(-1), N(2*i+1)) for i in 1:3])`